libs/designer-ui/src/lib/querybuilder/Group.tsx

```diff
@@ -183,7 +183,7 @@ export const Group = ({
   ];
 
   const handleUpdateNewParent = (newState: GroupItemProps | RowItemProps, currIndex: number) => {
-    const newItems = { ...groupProps };
+    const newItems = { ...groupProps, items: [...groupProps.items] };
```

Comment:
NIT: can this be `items: groupProps.items` since we are destructuring the array and creating a new array at the same time? or maybe I don't see a specific scenarios

Reply:
This causes the array to be non-extensible which prevents elements from being added, it runs into the error of:
`Cannot add property 0, object is not extensible` and is unable to add any changes the user mades